models/cut_model.py

```diff
@@ -166,21 +173,45 @@ def __init__(self, opt, rank):
                 betas=(opt.train_beta1, opt.train_beta2),
             )
             if opt.D_netD_global == "none":
-                self.optimizer_D = opt.optim(
-                    opt,
-                    self.netD.parameters(),
-                    lr=opt.train_D_lr,
-                    betas=(opt.train_beta1, opt.train_beta2),
-                )
+                if opt.D_temporal:
```

Comment:
Could we move all this logic by one level in the class hierarchy, e.g. into BaseModel ?